presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduDistributedQueries.java

```diff
@@ -0,0 +1,206 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kudu;
+
+import io.prestosql.testing.AbstractTestDistributedQueries;
+import io.prestosql.testing.MaterializedResult;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.testing.assertions.Assert;
+import io.prestosql.testing.sql.TestTable;
+import io.prestosql.tpch.TpchTable;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+
+import java.util.Optional;
+
+import static io.prestosql.plugin.kudu.KuduQueryRunnerFactory.createKuduQueryRunnerTpch;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.testing.MaterializedResult.resultBuilder;
+
+public class TestKuduDistributedQueries
+        extends AbstractTestDistributedQueries
+{
+    private TestingKuduServer kuduServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        kuduServer = new TestingKuduServer();
+        return createKuduQueryRunnerTpch(kuduServer, Optional.of(""), TpchTable.getTables());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        throw new SkipException("Kudu connector does not support column default values");
+    }
+
+    // TODO : Support these test once we are support create tables with default partitions
+
+    @Override
+    public void testInsert()
+    {
+    }
+
+    @Override
+    public void testDataMappingSmokeTest(DataMappingTestSetup dataMappingTestSetup)
+    {
+    }
+
+    @Override
+    public void testAddColumn()
+    {
+    }
+
+    @Override
+    public void testCreateTable()
+    {
+    }
+
+    @Override
+    public void testInsertUnicode()
+    {
+    }
+
+    @Override
+    public void testInsertWithCoercion()
+    {
+        // Override because of non-canonical varchar mapping
+    }
+
+    @Override
+    protected boolean supportsViews()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsArrays()
+    {
+        return false;
+    }
+
+    // TODO https://github.com/prestosql/presto/pull/2843
+
+    @Override
+    public void testScalarSubquery()
+    {
+    }
+
+    @Override
+    public void testScalarSubqueryWithGroupBy()
+    {
+    }
+
+    @Override
+    public void testCorrelatedJoin()
+    {
+    }
+
+    @Override
+    public void testCorrelatedNonAggregationScalarSubqueries()
+    {
+    }
+
+    @Override
+    public void testCorrelatedScalarSubqueries()
+    {
```

Comment:
What is the issue with subqueries?

Reply:
The actual pruned column list are not propagated to the ConnectorMetadata#applyProjection.  https://github.com/prestosql/presto/pull/2843 would ensure that the actual columns will be passed to the connector.

Reply:
Can you add a TODO comment that these tests should be enabled once above mentioned pull request is merged?

Reply:
i think the TODO should reference an _issue_ rather than a PR, even if PR is open.

should this be "TODO https://github.com/prestosql/presto/issues/3170"?